src/app/controllers/public-forms.server.controller.js

```diff
@@ -44,9 +44,19 @@ exports.isFormPublic = function (req, res, next) {
  * @param  {Object} res - Express response object
  */
 exports.redirect = async function (req, res) {
+  const queryPortion = req.url ? req.url.split('?', 2)[1] : undefined
```

Comment:
how about using `req.query` directly?

Reply:
@liangyuanruo considered this, but req.query returns an object, and that means order of the query params may not be retained

the implication is that upon redirect (i.e. adding the hashbang), the user who wants to eyeball the url to make sure it is still correct may find it difficult to compare with the expected string

if this is not a major concern, then will use req.query

Reply:
actually on further consideration it's also less complicated to pass the query string as is to the redirect, with res.query we are reconstructing the url string from the query object - thoughts?

Reply:
query parameters isn't supposed to have inherent order as well, so this shouldn't be an issue

Reply:
Agree, just use `req.query`.